solidity/test/token_grant/TestManagedGrantFactory.js

```diff
@@ -108,6 +108,25 @@ describe('TokenGrant/ManagedGrantFactory', () => {
       expect(event.args['grantee']).to.equal(grantee);
     });
 
+    it("works with receiveApproval", async () => {
+      grantStart = await time.latest();
+      let extraData = web3.eth.abi.encodeParameters(
+        ['address', 'uint256', 'uint256', 'uint256', 'bool'],
+        [grantee, grantUnlockingDuration.toNumber(), grantStart.toNumber(), grantCliff.toNumber(), false]
+      );
+      await token.approveAndCall(
+        factory.address, grantAmount, extraData, {from: grantCreator}
+      );
+      let event = (await factory.getPastEvents())[0];
+      expect(event.args['grantee']).to.equal(grantee);
+      let managedGrantAddress = event.args['grantAddress'];
+      let managedGrant = await ManagedGrant.at(managedGrantAddress);
+      let grantId = await managedGrant.grantId();
+      expect(await tokenGrant.availableToStake(grantId)).to.eq.BN(grantAmount);
+      expect(await managedGrant.grantee()).to.equal(grantee);
+      expect(await managedGrant.grantManager()).to.equal(grantCreator);
```

Comment:
Do you think it makes sense to assert cliff, duration, and start parameter of the created grant? They are all `uint256`, it's easy to swap them and not notice.